xarray/namedarray/core.py

```diff
@@ -847,5 +856,118 @@ def _to_dense(self) -> Self:
         else:
             raise TypeError("self.data is not a sparse array")
 
+    def transpose(
+        self,
+        *dims: Hashable | ellipsis,
+        missing_dims: ErrorOptionsWithWarn = "raise",
+    ) -> Self:
+        """Return a new object with transposed dimensions.
+
+        Parameters
+        ----------
+        *dims : Hashable, optional
+            By default, reverse the dimensions. Otherwise, reorder the
+            dimensions to this order.
+        missing_dims : {"raise", "warn", "ignore"}, default: "raise"
+            What to do if dimensions that should be selected from are not present in the
+            NamedArray:
+            - "raise": raise an exception
+            - "warn": raise a warning, and ignore the missing dimensions
+            - "ignore": ignore the missing dimensions
+
+        Returns
+        -------
+        transposed : NamedArray
+            The returned object has transposed data and dimensions with the
+            same attributes as the original.
+
+        Notes
+        -----
+        This operation returns a view of this variable's data. It is
+        lazy for dask-backed Variables but not for numpy-backed Variables.
+
+        See Also
+        --------
+        numpy.transpose
+        """
+        from xarray.core.indexing import as_indexable
+
+        if not dims:
+            dims = self.dims[::-1]
+        else:
+            dims = tuple(infix_dims(dims, self.dims, missing_dims))
+
+        if len(dims) < 2 or dims == self.dims:
+            # no need to transpose if only one dimension
+            # or dims are in same order
+            return self.copy(deep=False)
```

Comment:
Not a fan of this 1d array copy, it's a time thief. 

Array api suggests to raise an error instead. 
https://data-apis.org/array-api/latest/API_specification/generated/array_api.array.T.html

Reply:
happy to make this adjustment (raise an error `if len(dims) < 2`) if we deem it necessary. it appears it would result in breaking backward compatibility though.

Reply:
Yeah, I'm of the opinion that we shouldn't worry too much about backwards compatibility in NamedArray. Just handle the backwards compat in Variable instead.

Reply:
This is `permute_dims` though.